### PR TITLE
Optimizations here and here

### PR DESCRIPTION
**Pull request by SuperLeaf1995**
some smol optimizations

### CODE REVIEW

**Comment by segfaultdev** [comment]
Don't add `lseek` to devices, they have offsets passed to read and write functions already.

**Comment by segfaultdev** [comment]
Please fix memset and memcpy to use string operations(at least), set the serial buffer size to zero like it was before and DON'T TOUCH DEVICES. Thanks (:.

File: src/include/string.h [hunk 1]
```diff
@@ -6,11 +6,11 @@
 #include <stddef.h>
 #include <stdint.h>
 
-void *memset(void *b, int c, int len);
+void *memset(void *b, int c, size_t len);
 unsigned int strlen(const char *s);
 char *strcat(char *s1, const char *s2);
 char *strcpy(char *destination, const char *source);
-char *itoa(int value, char *str, int base);
+char *itoa(int value, char *str, size_t base);
```
Comment (unsafecast) [inline src/include/string.h:13]: Why did you change this? I don't see the point; and in the standard library it's an `int`

File: src/misc/string.c [hunk 2]
```diff
@@ -15,17 +37,32 @@ void *memset(void *b, int c, int len) {
...
+  size_t j = n % sizeof(uintptr_t);
+  while (j != 0) {
+    *cdest = *csrc;
+    cdest++;
+    csrc++;
+  }
 }
 
-unsigned int strlen(const char *s) {
+inline unsigned int strlen(const char *s) {
```
Comment (unsafecast) [inline src/misc/string.c:65]: This one though should return a `size_t`, as it does in the standard library
